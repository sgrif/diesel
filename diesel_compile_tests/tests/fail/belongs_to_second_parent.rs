#[macro_use]
extern crate diesel;

table! {
    bar (id) {
        id -> Integer,
    }
}

table! {
    foo (bar_id) {
        bar_id -> Integer,
    }
}

#[derive(Identifiable)]
#[diesel(table_name = bar)]
struct Bar {
    id: i32,
}

#[derive(Identifiable)]
#[diesel(table_name = bar)]
struct Baz {
    id: i32,
}

#[derive(Associations)]
#[diesel(belongs_to(Bar, Baz))]
#[diesel(table_name = foo)]
struct Foo {
    bar_id: i32,
}

fn main() {
    // Workaround for https://github.com/dtolnay/trybuild/issues/8
    compile_error!("warnings");
}
