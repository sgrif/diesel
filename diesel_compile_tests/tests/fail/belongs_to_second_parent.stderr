error: expected `=`
  --> $DIR/belongs_to_second_parent.rs:29:29
   |
29 | #[diesel(belongs_to(Bar, Baz))]
   |                             ^

error: warnings
  --> $DIR/belongs_to_second_parent.rs:37:5
   |
37 |     compile_error!("warnings");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
